File: test-samples/javascript/test-unused.js
// Test file for unused variable detection

const unusedVar = "this is never used";
let anotherUnused = 123;

function usedFunction() {
    return "used";
}

function unusedFunction() {
    return "not called anywhere";
}

const used = usedFunction();
console.log(used);

// Variable with underscore prefix should be ignored
const _ignoreThis = "should not trigger warning";

// Used variable
const active = true;
if (active) {
    console.log("active is true");
}

File: test-samples/dart/test-unused.dart
// Test file for unused variables and code

void main() {
  // Unused variables
  int unusedVariable = 42;
  String unusedString = "never used";
  double unusedDouble = 3.14;
  
  // Used variable (should not warn)
  int usedVariable = 10;
  print(usedVariable);
  
  // Unused function parameter
  calculateSum(5, 10);
  
  // Unused imports would be detected here
  var list = [1, 2, 3];
  
  // Unused local variable in block
  {
    int blockVar = 100;
    int usedBlockVar = 200;
    print(usedBlockVar);
  }
}

void calculateSum(int a, int b) {
  // Parameter 'b' is unused
  print(a);
}

class UnusedClass {
  int unusedField = 0;
  int usedField = 1;
  
  void unusedMethod() {
    print("never called");
  }
  
  void usedMethod() {
    print(usedField);
  }
}

void testFunction() {
  var obj = UnusedClass();
  obj.usedMethod();
}
